base/lock.jl

```diff
@@ -9,6 +9,28 @@ Get the currently running [`Task`](@ref).
 """
 current_task() = ccall(:jl_get_current_task, Ref{Task}, ())
 
+# This bit is set in the `havelock` of a `ReentrantLock` when that lock is locked by some task.
+const LOCKED_BIT = 0b01
+# This bit is set in the `havelock` of a `ReentrantLock` just before parking a task. A task is being
+# parked if it wants to lock the lock, but it is currently being held by some other task.
+const PARKED_BIT = 0b10
+
+const MAX_SPINS = 10
```

Comment:
This is in `log2`? The optimal number of spins is chip dependent if I recall the WTF code correctly.

Reply:
Yes, will change to `MAX_SPINS_LOG2`.

WTF definitely goes the extra mile of optimizing the spin limit, but `parking_lot` seems to be perform fine with a [hardcoded 10](https://github.com/Amanieu/parking_lot/blob/ca920b31312839013b4455aba1d53a4aede21b2f/core/src/spinwait.rs#L67) so I just used what they settled on.

Reply:
Actually, the number of spins is 10 (cmpxchg), its just the amount of work per spin that increases in each iteration to introduce some backoff.